src/note.cpp

```diff
@@ -459,6 +459,47 @@ TransPitch Note::GetTransPitch()
     return TransPitch(pname, this->GetChromaticAlteration(), this->GetOct());
 }
 
+void Note::UpdateFromTransPitch(TransPitch tp)
+{
+    data_PITCHNAME pname = static_cast<data_PITCHNAME>(tp.m_pname + PITCHNAME_c);
+    this->SetPname(pname);
+
+    Accid *accid = this->GetDrawingAccid();
+    if (accid->HasAccidGes()) {
+        data_ACCIDENTAL_GESTURAL gestural = ACCIDENTAL_GESTURAL_NONE;
+        switch (tp.m_accid) {
+            case -2: gestural = ACCIDENTAL_GESTURAL_ff; break;
+            case -1: gestural = ACCIDENTAL_GESTURAL_f; break;
+            case 0: gestural = ACCIDENTAL_GESTURAL_NONE; break;
+            case 1: gestural = ACCIDENTAL_GESTURAL_s; break;
+            case 2: gestural = ACCIDENTAL_GESTURAL_ss; break;
+            default: break; // TODO: Return an error here.
+        }
+        accid->SetAccidGes(gestural);
+    }
+    else {
+        data_ACCIDENTAL_WRITTEN written = ACCIDENTAL_WRITTEN_NONE;
+        switch (tp.m_accid) {
+            case -3: written = ACCIDENTAL_WRITTEN_tf; break;
+            case -2: written = ACCIDENTAL_WRITTEN_ff; break;
+            case -1: written = ACCIDENTAL_WRITTEN_f; break;
+            case 0: written = ACCIDENTAL_WRITTEN_NONE; break;
```

Comment:
I think if a **written** accidental was specified (a sharp, for example), we would want to write a natural if the transposed pitch had no Chromatic Alteration.

Reply:
Yes, I am pretty sure that a visual accidental will need to remain visible provided the key signature is transposed correctly at the same time.  If the key signature is not transposed by the same amount as the notes, then the accidentals can migrate between `@accid` and `@accid.ges` (and that is much more complicated to sort out, particularly with cross-staff notes and cautionary/courtesy accidentals that are not labeled as such).